chore: use deployment type in the deploy function (#1086)
matter-labs/zksync-web-era-docs: docs/build/tooling/hardhat/hardhat-zksync-deploy.md

:::

> :::note salt and deployment type within the deploy method
> To specify the `ContractDeployer` system contract function to be called, you need to provide the `deploymentType` parameter. If this parameter is omitted, the `create` function will be used as the default.
> 
> If salt is needed provide value through the overrides parameter. If the salt is omitted, the default value will be `0x0000000000000000000000000000000000000000000000000000000000000000`.
> 
> ```typescript
> const wallet = new zk.Wallet("PRIVATE_KEY");
> const deployer = new Deployer(hre, zkWallet);
> 
> const contract = await deployer.deploy("ContractName", [...args], "create2", {
>   customData: {
>     salt: "0x7935910912126667836566922594852029127629416664760357073852948630",
>   },
> });
> ```
> 
> :::
> 

Reviewer: ```suggestion
:::info salt and deployment type within the deploy method
You can provide the `deploymentType` parameter to specify the `ContractDeployer` system contract function that will be used. If this parameter is omitted, the `create` function will be used as the default. Allowed options are `create`, `create2`, `createAccount` and `create2Account`.

If salt is needed, provide its value in the overrides parameter. If the salt is omitted, the default value will be `0x0000000000000000000000000000000000000000000000000000000000000000`.

```typescript
const wallet = new zk.Wallet("PRIVATE_KEY");
const deployer = new Deployer(hre, zkWallet);

const contract = await deployer.deploy("ContractName", [...args], "create2", {
  customData: {
    salt: "0x7935910912126667836566922594852029127629416664760357073852948630",
  },
});
```

:::

```

];
```

- `--no-compile`- skip the compilation process, e.g. `hardhat deploy-zksync:contract --contract-name Contract --no-compile`.

> - `--deployment-type` - specify which deployer smart contract function will be called. Permissible values for this parameter include `create`, `create2`, `createAccount`, and `create2Account`. If this parameter is omitted, the default value will be `create`, e.g. `hardhat deploy-zksync:contract --contract-name Greeter 'Hello' --deployment-type create2`.

Reviewer: ```suggestion
- `--deployment-type` - specify which deployer smart contract function will be called. Allowed values for this parameter include `create`, `create2`, `createAccount`, and `create2Account`. If this parameter is omitted, the default value will be `create`, e.g. `hardhat deploy-zksync:contract --contract-name Greeter 'Hello' --deployment-type create2`.
```
